src/loaders/VolumeLoaderUtils.ts

```diff
@@ -104,6 +105,33 @@ export function estimateLevelForAtlas(spatialDimsZYX: number[][], maxAtlasEdge =
   return levelToLoad;
 }
 
+/** Given the size of a volume in pixels, convert a `Box3` in the 0-1 range to pixels */
+export function convertSubregionToPixels(region: Box3, size: Vector3): Box3 {
+  const min = region.min.clone().multiply(size).floor();
+  const max = region.max.clone().multiply(size).ceil();
+
+  // ensure it's always valid to specify the same number at both ends and get a single slice
+  if (min.x === max.x && min.x < size.x) {
+    max.x += 1;
+  }
+  if (min.y === max.y && min.y < size.y) {
+    max.y += 1;
+  }
+  if (min.z === max.z && min.z < size.z) {
+    max.z += 1;
+  }
+
+  return new Box3(min, max);
+}
+
+/** Transform a `Box3` in the 0-1 range into the space of a containing `Box3` */
```

Comment:
I'm not totally clear on what this does.  Is the containing Box3 in the same 0-1 coordinate space or in pixels?

Reply:
As used, both `region` and `container` are within 0-1, though only `region` has to be. This is basically getting the subregion of `container` represented by `region`. (Perhaps that should be the comment? idk why I've had so much trouble finding a clear docstring for this function, either here or in its previous form)

Reply:
i.e. if `container`'s range along the X axis is 0-4 and `region`'s is 0.25-0.5, the result will have range 1-2.

Reply:
``` Return the subset of container specified by assuming region contains fractional values. i.e. if container's range along the X axis is 0-4 and region's is 0.25-0.5, the result will have range 1-2.```

Reply:
(suggested docstring comment above)